src/AbstractFactory/ReflectionBasedAbstractFactory.php

```diff
@@ -105,20 +107,25 @@ public function __construct(array $aliases = [])
 
     /**
      * {@inheritDoc}
-     *
-     * @return DispatchableInterface
      */
-    public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null)
+    public function __invoke(ContainerInterface $container, string $requestedName, ?array $options = null): mixed
     {
+        if (! class_exists($requestedName)) {
+            throw new InvalidArgumentException(sprintf('%s can only be used with class names.', self::class));
+        }
```

Comment:
IMO this runtime check can be removed, replaced with:

 1. an `@param class-string $requestedName`
 2. a suppression (we obviously aren't honoring the parent type)

`new ReflectionClass($requestedName)` already performs this check for us anyway.

Reply:
This would open a huge can of worms:

1. `FactoryInterface` does not state `$requestedName` to be `class-string` as factories can return whatever
2. If I annotate `class-string` for the `ReflectionBasedAbstractFactory`, I also would've to suppress `MoreSpecificImplementedParamType`
3. After I've done that, I can remove the `class_exists` check

But for what? All factories are called from within the `ServiceManager`, there is no static analysis (yet) which prevents a user to actually use the factory with interfaces (even tho, that will lead to runtime errors).

So I'd rather keep the runtime check and provide a proper error rather than adding suppressions.